react/react1/week3/to-do-list-3/src/components/to-do-list.jsx

```diff
@@ -0,0 +1,300 @@
+import { useState } from "react";
+import { useEffect } from "react";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import BorderBox from "./border-box";
+import "./css/to-do-list.css";
+
+function ToDoList() {
+  const [toDoTaskTxtBox, setToDoTask] = useState("");
+  const [deadlineDate, setDeadlineDate] = useState(new Date());
+  const [dateErrorMsg, setDateErrorMsg] = useState("");
+  const [taskNameErrorMsg, setTaskNameErrorMsg] = useState("");
+  const [toDoList, setToDoList] = useState([]);
+
+  function inputTextChangeHandler(e) {
+    setToDoTask(e.target.value.trim());
+  }
+
+  function dateChangeHandler(date) {
+    const currentDate = new Date();
+    if (date <= currentDate) {
+      console.log("past date");
+      setDateErrorMsg("Pls enter a future date");
+    } else {
+      setDeadlineDate(date);
+      console.log(date);
+      setDateErrorMsg("");
+    }
+  }
+
+  function AddClickHandler() {
```

Comment:
Only capitalize the first letter if you are making components in React don't do it with regular functions, otherwise when reading the code a reader might think it is a component and get confused. Once again for better maintainability.